xbmc/video/dialogs/GUIDialogVideoVersion.cpp

```diff
@@ -0,0 +1,1008 @@
+  m_database.GetDefaultVideoVersion(itemType, dbId, *m_defaultVideoVersion.get());
+
+  CONTROL_DISABLE(CONTROL_BUTTON_SET_DEFAULT);
+}
+
+void CGUIDialogVideoVersion::SetDefaultVideoVersion(CFileItem& version)
+{
+  const int dbId = m_videoItem->GetVideoInfoTag()->m_iDbId;
+  VideoDbContentType itemType = m_videoItem->GetVideoContentType();
+
+  // set the specified video version as default
+  m_database.SetDefaultVideoVersion(itemType, dbId, version.GetVideoInfoTag()->m_iDbId);
+
+  // update the video item
+  m_videoItem->SetPath(version.GetPath());
+  m_videoItem->SetDynPath(version.GetPath());
+
+  // update video details since we changed the video file for the item
+  m_database.GetDetailsByTypeAndId(*m_videoItem, itemType, dbId);
+
+  // notify all windows to update the file item
+  CGUIMessage msg(GUI_MSG_NOTIFY_ALL, 0, 0, GUI_MSG_UPDATE_ITEM, GUI_MSG_FLAG_FORCE_UPDATE,
+                  m_videoItem);
+  CServiceBroker::GetGUI()->GetWindowManager().SendMessage(msg);
+}
+
+void CGUIDialogVideoVersion::ChooseArt()
+{
+  if (!CGUIDialogVideoInfo::ChooseAndManageVideoItemArtwork(
+          std::make_shared<CFileItem>(*m_selectedVideoVersion)))
+    return;
+
+  // update the thumbnail
+  CGUIControl* control = GetControl(CONTROL_IMAGE_THUMB);
+  if (control)
+  {
+    CGUIImage* imageControl = static_cast<CGUIImage*>(control);
+    imageControl->FreeResources();
+    imageControl->SetFileName(m_selectedVideoVersion->GetArt("thumb"));
+  }
+
+  CGUIMessage msg(GUI_MSG_NOTIFY_ALL, 0, 0, GUI_MSG_REFRESH_THUMBS);
+  CServiceBroker::GetGUI()->GetWindowManager().SendMessage(msg);
+}
+
+void CGUIDialogVideoVersion::AddVersion()
+{
+  AddVideoVersion(true);
+}
+
+void CGUIDialogVideoVersion::AddExtras()
+{
+  AddVideoVersion(false);
+}
+
+void CGUIDialogVideoVersion::AddVideoVersion(bool primary)
+{
+  const int dbId = m_videoItem->GetVideoInfoTag()->m_iDbId;
+  MediaType mediaType = m_videoItem->GetVideoInfoTag()->m_type;
+  VideoDbContentType itemType = m_videoItem->GetVideoContentType();
+
+  std::string title = primary ? StringUtils::Format(g_localizeStrings.Get(40014),
+                                                    CMediaTypes::GetLocalization(mediaType))
+                              : StringUtils::Format(g_localizeStrings.Get(40015),
+                                                    CMediaTypes::GetLocalization(mediaType));
+
+  // prompt to choose a video file
+  VECSOURCES sources = *CMediaSourceSettings::GetInstance().GetSources("files");
+
+  CServiceBroker::GetMediaManager().GetLocalDrives(sources);
+  CServiceBroker::GetMediaManager().GetNetworkLocations(sources);
+
+  std::string path;
+  if (CGUIDialogFileBrowser::ShowAndGetFile(
+          sources, CServiceBroker::GetFileExtensionProvider().GetVideoExtensions(), title, path))
+  {
+    std::string typeVideoVersion;
+    std::string videoTitle;
+    int idFile;
+    int idMedia;
+    MediaType itemMediaType;
+    VideoVersionItemType versionItemType;
```

Comment:
I know these are out parameters, but what speaks against initializing these vars? In theory, implementation of `m_database.GetVideoVersionInfo` could change in the future and then you have here another place for potential and hard to track down bugs (due to read of un-itited vars). If you init the vars although not currently really needed, there is one place less for obscure things that can happen in the future.